src/frontend/servers/y-provider/src/server.ts

```diff
@@ -1,18 +1,142 @@
+// eslint-disable-next-line import/order
+import './services/sentry';
 import { Server } from '@hocuspocus/server';
+import * as Sentry from '@sentry/node';
+import express, { Request, Response } from 'express';
+import expressWebsockets from 'express-ws';
 
-const port = Number(process.env.PORT || 4444);
+import { PORT } from './env';
+import { httpSecurity, wsSecurity } from './middlelayers';
+import { routes } from './routes';
+import { logger } from './utils';
 
-const server = Server.configure({
-  name: 'docs-y-provider',
-  port: port,
+export const hocuspocusServer = Server.configure({
+  name: 'docs-y-server',
   timeout: 30000,
-  debounce: 2000,
-  maxDebounce: 30000,
   quiet: true,
-});
+  onConnect({ requestHeaders, connection, documentName, requestParameters }) {
+    const roomParam = requestParameters.get('room');
+    const canEdit = requestHeaders['x-can-edit'] === 'True' ? true : false;
+
+    if (!canEdit) {
+      connection.readOnly = true;
+    }
+
+    logger(
+      'Connection established:',
+      documentName,
+      'userId:',
+      requestHeaders['x-user-id'],
+      'canEdit:',
+      canEdit,
+      'room:',
+      requestParameters.get('room'),
+    );
+
+    if (documentName !== roomParam) {
+      console.error(
+        'Invalid room name - Probable hacking attempt:',
```

Comment:
Why do you need the room parameter if you already have the info in documentName? :thinking:

Reply:
`roomParam` is needed for 2 things:
- we extract the document from the request_uri in `collaboration-auth` endpoint -  ([here](https://github.com/numerique-gouv/impress/pull/472/files#diff-cc27a5de6f1d8131250e4be30d0fd4c2f1a816bdc35951a9c86e6f13cd795019R49))
- We need it to stick to the good pod between clients
https://github.com/numerique-gouv/impress/pull/472/files#diff-259585b31c6635e6f972a090c26e576f7ae41f7cc7d3f65be7cf16c08fbd2f2fR80